operator/internal/manifests/gateway_tenants.go

```diff
@@ -175,3 +193,64 @@ func ConfigureOptionsForMode(cfg *config.Options, opt Options) error {
 
 	return nil
 }
+
+// configureMTLS will mount CA bundles and fix CLI arguments for the gateway container
+// if any tenant configured mTLS authentication
+func configureMTLS(d *appsv1.Deployment, tenants *lokiv1.TenantsSpec) error {
+	var gwIndex int
+	for i, c := range d.Spec.Template.Spec.Containers {
+		if c.Name == gatewayContainerName {
+			gwIndex = i
+			break
+		}
+	}
+
+	gwContainer := d.Spec.Template.Spec.Containers[gwIndex].DeepCopy()
+	gwArgs := gwContainer.Args
+	gwVolumes := d.Spec.Template.Spec.Volumes
+
+	mTLS := false
+	for _, tenant := range tenants.Authentication {
+		if tenant.MTLS != nil {
+			gwContainer.VolumeMounts = append(gwContainer.VolumeMounts, corev1.VolumeMount{
+				Name:      tenantMTLSVolumeName(tenant.TenantName),
+				MountPath: tenantMTLSCADir(tenant.TenantName),
+			})
+			gwVolumes = append(gwVolumes, corev1.Volume{
+				Name: tenantMTLSVolumeName(tenant.TenantName),
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: tenant.MTLS.CASpec.CA,
+						},
+					},
+				},
+			})
+			mTLS = true
+		}
+	}
+	if !mTLS {
+		return nil // nothing to configure
+	}
+
+	// Remove old tls.client-auth-type
+	for i, arg := range gwArgs {
+		if strings.HasPrefix(arg, "--tls.client-auth-type=") {
+			gwArgs = append(gwArgs[:i], gwArgs[i+1:]...)
+			break
+		}
+	}
+	gwArgs = append(gwArgs, "--tls.client-auth-type=RequestClientCert")
+
+	gwContainer.Args = gwArgs
+	p := corev1.PodSpec{
+		Containers: []corev1.Container{
+			*gwContainer,
+		},
+		Volumes: gwVolumes,
+	}
+	if err := mergo.Merge(d.Spec.Template.Spec, p, mergo.WithOverride); err != nil {
```

Comment:
```suggestion
	if err := mergo.Merge(&d.Spec.Template.Spec, p, mergo.WithOverride); err != nil {
```
:point_up: Requires the destination to be a point in order to merge.